news-analysis.py

```diff
@@ -443,15 +455,18 @@ def sell(compiled_sentiment, headlines_analysed):
                 # retrieve the last order
                 order = client.get_all_orders(symbol=coin+PAIRING, limit=1)
 
-                # convert order timsestamp into UTC format
-                time = order[0]['time'] / 1000
-                utc_time = datetime.fromtimestamp(time)
+                if order:
+                    # convert order timsestamp into UTC format
+                    time = order[0]['time'] / 1000
+                    utc_time = datetime.fromtimestamp(time)
 
-                # grab the price of CRYPTO the order was placed at for reporting
-                sold_at = CURRENT_PRICE[coin+PAIRING]
+                    # grab the price of CRYPTO the order was placed at for reporting
+                    sold_at = CURRENT_PRICE[coin+PAIRING]
 
-                # print order condirmation to the console
-                print(f"order {order[0]['orderId']} has been placed on {coin} with {order[0]['origQty']} coins sold for {sold_at} each at {utc_time}")
+                    # print order condirmation to the console
+                    print(f"order {order[0]['orderId']} has been placed on {coin} with {order[0]['origQty']} coins sold for {sold_at} each at {utc_time}")
```

Comment:
general comment, seems that there is a lot of code duplication between sell and buy methods. At least the logging can be moved to a separate method.